tools/cli/factory.go

```diff
@@ -139,6 +135,31 @@ func (b *clientFactory) ServerAdminClient(c *cli.Context) admin.Client {
 	return admin.NewThriftClient(serverAdmin.New(clientConfig))
 }
 
+// ServerFrontendClientForMigration builds a frontend client (based on server side thrift interface)
+func (b *clientFactory) ServerFrontendClientForMigration(c *cli.Context) frontend.Client {
```

Comment:
the code is exactly the same as `ServerFrontendClient`, should `dispatcher` be replaced with something else?

Reply:
good catch. fixed